### ISSUE TITLE
Enhancement: `consistent-type-imports` should have an option to prevent cases with only inline type imports

### ISSUE THREAD

**Issue by me4502** [typescript-eslint/typescript-eslint#6379]
### Before You File a Proposal Please Confirm You Have Done The Following...

- [X] I have [searched for related issues](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aissue+label%3A%22enhancement%3A+plugin+rule+option%22) and found none that match my proposal.
- [X] I have searched the [current rule list](https://typescript-eslint.io/rules/#supported-rules) and found no rules that match my proposal.
- [X] I have [read the FAQ](https://typescript-eslint.io/linting/troubleshooting) and my problem is not listed.

### My proposal is suitable for this project

- [X] I believe my proposal would be useful to the broader TypeScript community (meaning it is not a niche proposal).

### Link to the rule's documentation

https://typescript-eslint.io/rules/consistent-type-imports/

### Description

TypeScript 5.0 is [introducing a new `--verbatimModuleSyntax`](https://devblogs.microsoft.com/typescript/announcing-typescript-5-0-beta/#verbatimmodulesyntax) flag/option, that enforces type imports to output exactly as they are written, rather than letting the compiler decide. One of the noted output conditions of this is that `import { type xyz } from "xyz";` will be rewritten to `import "xyz";`. This means inline type imports can and will still result in the file ending up imported. While this will always happen with this option enabled, it's also possible for it to happen currently if TypeScript decides to import it.

As this introduces potential unwanted imports, I propose a new option to disallow imports that purely contain inline type imports. The option could then add an auto-fixer to move the type import to the base of the import rather than inline. Eg, from `import { type a } from './xyz';` -> `import type { a } from './xyz';`

### Fail

```typescript
import { type a, type x } from './xyz';
```

### Pass

```typescript
import { a, type b } from './xyz';

import type { b } from './xyz';
```

**Comment by bradzacher** [comment]
`consistent-type-imports` is concerned with exactly one thing - ensuring your value imports are used as values, warning you when your value imports aren't used as values.

ESLint rules are best designed to handle one concern, rather than multiple.

What you're asking for here isn't related to ensuring type-only import specifiers are correctly annotated. Instead it's related to the semantics of how TS may elide import specifiers or import declarations based on the type specifier.

There are existing lint rules that can help out here, such as [`import/consistent-type-specifier-style`](https://github.com/import-js/eslint-plugin-import/blob/main/docs/rules/consistent-type-specifier-style.md).
